plugins/available/go.plugin.bash

```diff
@@ -1,19 +1,26 @@
-#!/usr/bin/env bash
-
+# shellcheck shell=bash
 cite about-plugin
 about-plugin 'go environment variables & path configuration'
 
-command -v go &>/dev/null || return
+# Load after basher and goenv
+# BASH_IT_LOAD_PRIORITY: 270
+
+{ _command_exists go && go version &> /dev/null; } || return 0
```

Comment:
why do you need the `go version` check here?

Reply:
> The nature of goenv allows for a situation where the initial shell does not point to a working go binary, and instead at the included shim script

This line sort of explains it. When using any of the *env tools, they create a "shim" script that does some environment prep before exec'ing the language tool, e.g. `go`. So even if the parent system doesn't have `go` actually installed, it will be found in PATH and be hashed for `type` lookups, but not be able to do anything with it:

```shell
$ go
goenv: 'go' command not found

The 'go' command exists in these Go versions:
  1.15.2
```

I can't use `go` without any sub-command because it will return `2`, so I test the version instead.

Reply:
@NoahGorny I've added a bunch of comments to better explain the thinking of the plugin.